server/swap/swap.go

```diff
@@ -1357,6 +1357,7 @@ func (s *Swapper) processInit(msg *msgjson.Message, params *msgjson.Init, stepIn
 		Time:     uint64(swapTimeMs),
 		CoinID:   params.CoinID,
 		Contract: params.Contract,
+		TxData:   contract.TxData(),
```

Comment:
Do you think ultimately the client can provide the raw tx in the init request to the server as well?  This PR hits the other half of it great though -- supplementing the server-originating audit and match_status requests, so that that client may rebroadcast it in subsequent work.

Reply:
With our current protocol and assets, [we have to fetch the transaction](https://github.com/decred/dcrdex/blob/5394ceaa2a0e69518cabb2a3ae4fdb2164a6a08e/server/asset/dcr/dcr.go#L794) anyway as part of validation. I like the idea of the client providing the details from a standpoint of limiting the server's role as a source of on-chain data, but it's net effect right now would just be more bandwidth. I could be convinced though, if you're adamant.

Reply:
Nope, just a thought.